### PR TITLE
[cmd/telemetrygen] Add support for metric name/attributes

### PR DESCRIPTION
**Pull request by claudiobastos**
**Description:** Add support to customize metric name and its attributes.

**Link to tracking Issue:** #26505 

**Testing:** -

**Documentation:** self-documented in added flags

### CODE REVIEW

File: cmd/telemetrygen/internal/common/config.go [hunk 1]
```diff
@@ -56,6 +56,8 @@ type Config struct {
 	HTTPPath           string
 	Headers            KeyValue
 	ResourceAttributes KeyValue
+	SignalAttributes   KeyValue
```
Comment (codeboten) [inline cmd/telemetrygen/internal/common/config.go:59]: As these attributes apply to the telemetry data being transmitted (metric datapoints, spans, log records) what would you think of the name `TelemetryAttributes` instead?

File: cmd/telemetrygen/internal/metrics/worker.go [hunk 2]
```diff
@@ -26,7 +27,7 @@ type worker struct {
 	index          int             // worker index
 }
 
-func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Exporter) {
+func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Exporter, name string, signalAttrs []attribute.KeyValue) {
```
Comment (codeboten) [inline cmd/telemetrygen/internal/metrics/worker.go:30]: the signal attributes should apply to all signals

File: cmd/telemetrygen/internal/metrics/worker.go [hunk 3]
```diff
@@ -37,12 +38,13 @@ func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Expor
...
-							Name: "gen",
+							Name: name,
 							Data: metricdata.Gauge[int64]{
 								DataPoints: []metricdata.DataPoint[int64]{
 									{
-										Time:  time.Now(),
-										Value: i,
+										Time:       time.Now(),
+										Value:      i,
+										Attributes: attribute.NewSet(signalAttrs...),
```
Comment (codeboten) [inline cmd/telemetrygen/internal/metrics/worker.go:47]: Please add unit tests to validate the changes

**Review by codeboten — COMMENTED** [review]
Thanks for the contribution @claudiobastos! Would you be open to splitting adding the metric name option from adding telemetry attributes? It would speed up going through the review process

**Review by codeboten — APPROVED** [review]
Thanks @claudiobastos, only one conflict needs resolving

**Comment by claudiobastos** [comment]
> Thanks @claudiobastos, only one conflict needs resolving
@codeboten 
Sorry, I didn't understand if I'm responsible to solve it or it's beyond my scope right now. Could you enlighten me ?

**Comment by codeboten** [comment]
@claudiobastos i gave it a shot, will see if i successfully merged things through the github UI without breaking anything 🤞🏻

The conflicting change that was merged was a documentation change.

File: cmd/telemetrygen/internal/common/config.go [hunk 4]
```diff
@@ -92,4 +104,8 @@ func (c *Config) CommonFlags(fs *pflag.FlagSet) {
 	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=\"value\"."+
 		"Note you may need to escape the quotes when using the tool from a cli."+
 		"Flag may be repeated to set multiple attributes (e.g -otlp-attributes key1=\"value1\" -otlp-attributes key2=\"value2\")")
+
+	c.TelemetryAttributes = make(map[string]string)
+	fs.Var(&c.TelemetryAttributes, "telemetry-attributes", "Custom telemetry attributes to use. The value is expected in the format \"key=\\\"value\\\"\". "+
+		"Flag may be repeated to set multiple attributes (e.g --signal-attributes \"key1=\\\"value1\\\"\" --otlp-attributes \"key2=\\\"value2\\\"\")")
```
Comment (atoulme) [inline cmd/telemetrygen/internal/common/config.go:110]: `signal-attributes` ? Did you mean `otlp-attributes`?

Comment (claudiobastos) [reply cmd/telemetrygen/internal/common/config.go:110]: sry, there was a typo. Both should be signal-attributes. This flag is to set telemetry signal attributes

**Review by mx-psi — APPROVED** [review]

**Comment by claudiobastos** [comment]
Oh... now we have more types for metrics \o/
I'll fix my PR asap

**Comment by claudiobastos** [comment]
Is there anything else I can do to improve this and allow merge ? @codeboten

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**